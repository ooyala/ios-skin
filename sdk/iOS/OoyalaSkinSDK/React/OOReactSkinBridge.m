//
//  OOReactSkinBridge.m
//  OoyalaSkinSDK
//
//  Created on 8/14/18.
//  Copyright Â© 2018 ooyala. All rights reserved.
//

#import "OOReactSkinBridge.h"
#import "OOReactSkinBridgeModule.h"
#import <React/RCTBridge+Private.h>

@interface OOReactSkinBridge ()

@property (nonatomic, weak, readonly) id<OOReactSkinBridgeDelegate> delegate;

@end


@implementation OOReactSkinBridge

@dynamic delegate;

- (instancetype)initWithDelegate:(id<OOReactSkinBridgeDelegate>)delegate
                   launchOptions:(NSDictionary *)launchOptions {
  return [super initWithDelegate:delegate launchOptions:launchOptions];
}

- (void)invalidate {
  [super invalidate];
  [NSNotificationCenter.defaultCenter removeObserver:self];
}

- (void)setUp {
  [super setUp];
  [NSNotificationCenter.defaultCenter addObserver:self
                                         selector:@selector(didInitializeModuleNotification:)
                                             name:RCTDidInitializeModuleNotification
                                           object:self.batchedBridge];
}

- (void)didInitializeModuleNotification:(NSNotification *)notification {
  id module = notification.userInfo[@"module"];

  if ([self.delegate respondsToSelector:@selector(bridge:didLoadModule:)]) {
    if (module && [module conformsToProtocol:@protocol(OOReactSkinBridgeModule)]) {
      [self.delegate bridge:self didLoadModule:module];
    }
  }
}

@end


@implementation RCTBridge (OOReactSkinEventsEmitterable)

- (OOReactSkinEventsEmitter *)skinEventsEmitter {
  return [self moduleForName:[OOReactSkinEventsEmitter moduleName]];
}

@end
