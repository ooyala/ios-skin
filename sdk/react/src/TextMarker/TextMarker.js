// @flow

import * as React from 'react';
import { Text, TouchableWithoutFeedback } from 'react-native';

import styles from './TextMarker.styles';

const TEXT_MARKER_COLLAPSED_LENGTH = 80;

type Props = {
  onSeek: () => void,
  text: string,
};

type State = {
  isExpanded: boolean,
};

export default class TextMarker extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    (this: Object).handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    const { onSeek, text } = this.props;
    const { isExpanded } = this.state;

    // Trigger seek callback if the marker text is shorter than maximum or has been expanded. If there the text is
    // longer we have to expand it first and only after the second click on that text trigger the callback.
    if (text.length <= TEXT_MARKER_COLLAPSED_LENGTH || isExpanded) {
      onSeek();
    }

    this.setState({
      isExpanded: !isExpanded,
    });
  }

  render() {
    const { text } = this.props;
    const { isExpanded } = this.state;

    let shownText = text;

    if (!isExpanded && text.length > TEXT_MARKER_COLLAPSED_LENGTH) {
      shownText = text.slice(0, TEXT_MARKER_COLLAPSED_LENGTH).concat('...');
    }

    return (
      <TouchableWithoutFeedback onPress={this.handlePress}>
        <Text style={styles.root} suppressHighlighting>{shownText}</Text>
      </TouchableWithoutFeedback>
    );
  }
}
