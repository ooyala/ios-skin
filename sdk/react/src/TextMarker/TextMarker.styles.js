// @flow

import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  root: {
    backgroundColor: 'red',
    bottom: 10,
    color: 'white',
    left: 0,
    position: 'absolute',
    width: 100,
  },
});
