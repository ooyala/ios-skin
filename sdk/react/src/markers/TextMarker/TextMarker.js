// @flow

import * as React from 'react';
import { Text, TouchableWithoutFeedback } from 'react-native';
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';

import styles from './TextMarker.styles';

const TEXT_MARKER_COLLAPSED_LENGTH = 8;

type Props = {
  backgroundColor?: ?string,
  onSeek: () => void,
  text: string,
  style?: ViewStyleProp,
};

type State = {
  isExpanded: boolean,
};

export default class TextMarker extends React.Component<Props, State> {
  static defaultProps = {
    backgroundColor: undefined,
    style: undefined,
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    (this: Object).handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    const { onSeek, text } = this.props;
    const { isExpanded } = this.state;

    // Trigger seek callback if the marker text is shorter than maximum or has been expanded. If there the text is
    // longer we have to expand it first and only after the second click on that text trigger the callback.
    if (text.length <= TEXT_MARKER_COLLAPSED_LENGTH || isExpanded) {
      onSeek();
    }

    this.setState({
      isExpanded: !isExpanded,
    });
  }

  render() {
    const { backgroundColor, style, text } = this.props;
    const { isExpanded } = this.state;

    let shownText = text;

    if (!isExpanded && text.length > TEXT_MARKER_COLLAPSED_LENGTH) {
      shownText = text.slice(0, TEXT_MARKER_COLLAPSED_LENGTH - 3).concat('...'); // 3 is the dots length
    }

    return (
      <TouchableWithoutFeedback onPress={this.handlePress}>
        <Text
          style={[
            styles.root,
            style,
            backgroundColor && { backgroundColor },
            isExpanded && styles.expanded,
          ]}
          suppressHighlighting
        >
          {shownText}
        </Text>
      </TouchableWithoutFeedback>
    );
  }
}
