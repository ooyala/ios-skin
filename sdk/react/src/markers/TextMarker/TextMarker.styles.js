// @flow

import { StyleSheet } from 'react-native';

const BORDER_WIDTH = 1;
const COLLAPSED_WIDTH = 64;
const EXPANDED_WIDTH = 128;
const FONT_SIZE = 13;
const PADDING = 2;

export default StyleSheet.create({
  expanded: {
    width: EXPANDED_WIDTH + 2 * PADDING + 2 * BORDER_WIDTH,
  },
  root: {
    backgroundColor: 'black',
    borderColor: '#4E4E4E',
    borderRadius: 2,
    borderWidth: BORDER_WIDTH,
    bottom: 6,
    color: 'white',
    fontSize: FONT_SIZE,
    lineHeight: FONT_SIZE,
    padding: PADDING,
    position: 'absolute',
    shadowColor: 'black',
    shadowOffset: { height: 2, width: 0 },
    shadowOpacity: 0.5,
    shadowRadius: 4,
    textAlign: 'center',
    width: COLLAPSED_WIDTH + 2 * PADDING + 2 * BORDER_WIDTH,
  },
});
