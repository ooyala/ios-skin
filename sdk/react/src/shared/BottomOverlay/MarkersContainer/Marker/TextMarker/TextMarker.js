// @flow

import * as React from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';

import { MARKERS_SIZES } from '../../../../../constants';
import styles from './TextMarker.styles';

const TEXT_MARKER_COLLAPSED_LENGTH = 8;

type Props = {
  backgroundColor?: ?string,
  onSeek: () => void,
  text: string,
  style?: ViewStyleProp,
};

type State = {
  isExpanded: boolean,
};

export default class TextMarker extends React.Component<Props, State> {
  static defaultProps = {
    backgroundColor: undefined,
    style: undefined,
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    (this: Object).handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    this.setState(({ isExpanded }, { onSeek, text }) => {
      // Trigger seek callback if the marker text is shorter than maximum or has been expanded. If there the text is
      // longer we have to expand it first and only after the second click on that text trigger the callback.
      if (text.length <= TEXT_MARKER_COLLAPSED_LENGTH || isExpanded) {
        onSeek();
      }

      return {
        isExpanded: !isExpanded,
      };
    });
  }

  render() {
    const { backgroundColor, style, text } = this.props;
    const { isExpanded } = this.state;

    let shownText = text;

    if (!isExpanded && text.length > TEXT_MARKER_COLLAPSED_LENGTH) {
      shownText = text.slice(0, TEXT_MARKER_COLLAPSED_LENGTH - 3).concat('...'); // 3 is the dots length
    }

    return (
      <TouchableWithoutFeedback onPress={this.handlePress}>
        <View
          style={[
            styles.root,
            style,
            backgroundColor && { backgroundColor },
            isExpanded && styles.expanded,
          ]}
        >
          <Text
            numberOfLines={MARKERS_SIZES.TEXT_NUMBER_OF_LINES}
            style={styles.text}
            suppressHighlighting
          >
            {shownText}
          </Text>
          <View style={[styles.triangle, backgroundColor && { borderTopColor: backgroundColor }]} />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}
