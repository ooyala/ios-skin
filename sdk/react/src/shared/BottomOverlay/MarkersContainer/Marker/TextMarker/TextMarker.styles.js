// @flow

import { StyleSheet } from 'react-native';

import { MARKERS_SIZES } from '../../../../../constants';
import commonMarkerStyle from '../styles/commonMarker.style';
import triangleStyle from '../styles/triangle.style';

const collapsedWidth = MARKERS_SIZES.TEXT_COLLAPSED_WIDTH + 2 * MARKERS_SIZES.PADDING + 2 * MARKERS_SIZES.BORDER_WIDTH;
const expandedWidth = MARKERS_SIZES.TEXT_EXPANDED_WIDTH + 2 * MARKERS_SIZES.PADDING + 2 * MARKERS_SIZES.BORDER_WIDTH;

export default StyleSheet.create({
  expanded: {
    transform: [
      { translateX: expandedWidth / -2 },
    ],
    width: expandedWidth,
  },
  root: {
    ...commonMarkerStyle,
    transform: [
      { translateX: collapsedWidth / -2 },
    ],
    width: collapsedWidth,
  },
  text: {
    color: 'white',
    fontSize: MARKERS_SIZES.FONT_SIZE,
    lineHeight: MARKERS_SIZES.FONT_SIZE,
    textAlign: 'center',
  },
  triangle: triangleStyle,
});
