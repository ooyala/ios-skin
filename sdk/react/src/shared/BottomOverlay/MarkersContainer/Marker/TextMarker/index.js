// @flow

import TextMarker from './TextMarker';

export default TextMarker;
