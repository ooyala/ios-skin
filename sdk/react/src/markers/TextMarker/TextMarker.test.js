// @flow

import React from 'react';
import TestRenderer from 'react-test-renderer';

import TextMarker from './TextMarker';

describe('TextMarker', () => {
  it('renders matching snapshot', () => {
    const testRenderer = TestRenderer.create(
      <TextMarker
        text="Hello, world"
        onSeek={() => undefined}
      />,
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
