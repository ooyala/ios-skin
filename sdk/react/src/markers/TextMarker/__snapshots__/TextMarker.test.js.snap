// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextMarker renders matching snapshot 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "borderColor": "#4E4E4E",
        "borderRadius": 2,
        "borderWidth": 1,
        "bottom": 6,
        "padding": 2,
        "position": "absolute",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
        "transform": Array [
          Object {
            "translateX": -35,
          },
        ],
        "width": 70,
      },
      undefined,
      undefined,
      false,
    ]
  }
>
  <Text
    numberOfLines={4}
    style={
      Object {
        "color": "white",
        "fontSize": 13,
        "lineHeight": 13,
        "textAlign": "center",
      }
    }
    suppressHighlighting={true}
  >
    Hello...
  </Text>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0,
          "borderColor": "transparent",
          "borderTopColor": "#4E4E4E",
          "borderWidth": 6,
          "bottom": -6,
          "height": 0,
          "left": "50%",
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": -3,
            },
          ],
          "width": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
